Tests/Rules/UseProcessBlockForPipelineCommands.tests.ps1

```diff
@@ -0,0 +1,37 @@
+﻿Describe "UseProcessBlockForPipelineCommands" {
+    BeforeAll {
+        $RuleName = 'PSUseProcessBlockForPipelineCommands'
+        $NoProcessBlock = 'function BadFunc1 { [CmdletBinding()] param ([Parameter(ValueFromPipeline)]$Param1) }'
+        $NoProcessBlockByPropertyName = 'function $BadFunc2 { [CmdletBinding()] param ([Parameter(ValueFromPipelineByPropertyName)]$Param1) }'
+        $HasProcessBlock = 'function GoodFunc1 { [CmdletBinding()] param ([Parameter(ValueFromPipeline)]$Param1) process { } }'
+        $HasProcessBlockByPropertyName = 'function GoodFunc2 { [CmdletBinding()] param ([Parameter(ValueFromPipelineByPropertyName)]$Param1) process { } }'
+        $NoAttribute = 'function GoodFunc3 { [CmdletBinding()] param ([Parameter()]$Param1) }'
+		$HasTypeDeclaration = 'function GoodFunc3 { [CmdletBinding()] param ([Parameter()][string]$Param1) }'
+    }
+
+    Context "When there are violations" {
+        $Cases = @(
+            @{ScriptDefinition = $NoProcessBlock; Name = "NoProcessBlock"}
+            @{ScriptDefinition = $NoProcessBlockByPropertyName; Name = "NoProcessBlockByPropertyName"}
+        )
+        It "has 1 violation for function <Name>" {
+            param ($ScriptDefinition)
+
+            Invoke-ScriptAnalyzer -ScriptDefinition $ScriptDefinition -IncludeRule $RuleName | Should Not BeNullOrEmpty
```

Comment:
Pester v4 syntax is some backwards compatibility with the v3 (syntax that does not have the dashes) but to stay on the save side, it is better to use the v4 syntax since we use v4.
```suggestion
            Invoke-ScriptAnalyzer -ScriptDefinition $ScriptDefinition -IncludeRule $RuleName | Should -Not -BeNullOrEmpty
```

Reply:
fixed in 2df5e700